modelling/src/main/java/org/axonframework/modelling/command/AbstractRepository.java

```diff
@@ -258,6 +281,22 @@ protected void reportIllegalState(A aggregate) {
      */
     protected abstract A doLoad(String aggregateIdentifier, Long expectedVersion);
 
+
+    /**
+     * Loads an aggregate from the reporsitory. If the aggregate does not exists, it is created using the {@code
+     * factoryMethod}.
+     *
+     * @param aggregateIdentifier the identifier of the aggregate
+     * @param expectedVersion     the expected version of the aggregate (null for no check)
+     * @param factoryMethod       the method that creates a new instance
+     * @return the aggregate
+     *
+     * @throws Exception when loading or creating the aggregate failed
+     */
+    protected A doLoadOrCreate(String aggregateIdentifier, Long expectedVersion, Callable<T> factoryMethod)
```

Comment:
I assume the implementation of this, which would likely call `doCreateNew` and `doLoad`, comes in a following PR?

Reply:
Implementation is in LockingRepostory.